src/istore_agg.c

```diff
@@ -3,6 +3,19 @@
 #define SAMESIGN(a,b)   (((a) < 0) == ((b) < 0))
 #define INITSTATESIZE 30
 
+#define AGG_TRANSFN_INIT                                                                     \
+    MemoryContext  agg_context;                                                              \
+    ISAggState    *state;                                                                    \
+                                                                                             \
+    if (!AggCheckCallContext(fcinfo, &agg_context))                                          \
+        elog(ERROR, "aggregate function called in non-aggregate context");                   \
+                                                                                             \
+    if (PG_ARGISNULL(1) && PG_ARGISNULL(0)) PG_RETURN_NULL();                                \
+                                                                                             \
+    state = PG_ARGISNULL(0) ? state_init(agg_context) : (ISAggState *) PG_GETARG_POINTER(0); \
+                                                                                             \
+    if (PG_ARGISNULL(1)) PG_RETURN_POINTER(state);                                           \
+
```

Comment:
it's probably best practice to add block though
```
#define AGG_TRANSFN_INIT \
{\
.... \
}
```
@roa or @zeitgeist can add their thoughts